### PR TITLE
Fix when Session, Authentication or Message middlewares are not present

### PR DESCRIPTION
**Pull request by mgaligniana**
Hello!

In this PR I try fix the issue https://github.com/jazzband/django-silk/issues/347

Doesn't fail when one of these middlewares are not present:
```
'django.contrib.sessions.middleware.SessionMiddleware',
'django.contrib.auth.middleware.AuthenticationMiddleware',
'django.contrib.messages.middleware.MessageMiddleware'
```

and if `SILKY_AUTHENTICATION` is `True`, an exception is raised: **SILKY_AUTHENTICATION can not be enabled without Session, Authentication or Message Django's middlewares**

Let me know any change required!

Thank you!

### CODE REVIEW

File: silk/middleware.py [hunk 1]
```diff
@@ -64,6 +71,9 @@ def process_request(self, request):
 
 class SilkyMiddleware:
     def __init__(self, get_response):
+        if config.SILKY_AUTHENTICATION and not (set(AUTH_AND_SESSION_MIDDLEWARES) & set(settings.MIDDLEWARE)):
+            raise SilkNotConfigured("SILKY_AUTHENTICATION can not be enabled without Session, Authentication or Message Django's middlewares")
+
         self.get_response = get_response
```
Comment (mgaligniana) [inline silk/middleware.py:77]: Not sure if it's the best place to do the check, but when line `self.get_response = get_response` runs the exception mentioned in the ticked is raised

**Comment by yoonthegoon** [comment]
`profiling.py`, `requests.py`, and `summary.py` look good to me. 
The tests look good at a glance.

Appears to resolve the issue on the ticket. 
Hopefully Seb or someone else comes in and checks it too.

**Review by albertyw — APPROVED** [review]

File: silk/views/requests.py [hunk 2]
```diff
@@ -130,7 +130,7 @@ def _get_objects(self, show=None, order_by=None, order_dir=None, path=None, filt
         return query_set[:show]
 
     def _create_context(self, request):
-        raw_filters = request.session.get(self.session_key_request_filters, {}).copy()
+        raw_filters = request.session.get(self.session_key_request_filters, {}).copy() if hasattr(request, 'session') else {}
```
Comment (SebCorbin) [inline silk/views/requests.py:133]: Thanks for your work, it's a great beginning, but if I'm not mistaking, the filters are not working when selectining them. They should work if you fall back to `request.GET or request.POST` here

**Review by SebCorbin — CHANGES_REQUESTED** [review]

File: silk/views/summary.py [hunk 3]
```diff
@@ -54,7 +54,7 @@ def _num_queries_by_view(self, filters):
         return sorted(requests, key=lambda item: item.t, reverse=True)
 
     def _create_context(self, request):
-        raw_filters = request.session.get(self.filters_key, {})
+        raw_filters = request.session.get(self.filters_key, {}) if hasattr(request, 'session') else {}
```
Comment (SebCorbin) [inline silk/views/summary.py:57]: And also a fallback to `request.GET or request.POST` here

**Review by SebCorbin — CHANGES_REQUESTED** [review]

**Comment by mgaligniana** [comment]
@SebCorbin First, thank you for your quick responses and the pacience 🙏 

Regarding the review:

> Thanks for your work, it's a great beginning, but if I'm not mistaking, the filters are not working when selecting them. They should work if you fall back to `request.GET or request.POST` here

I followed the `session` middleware variable approach and defined a new one for the silk's filters: [silk_filters](https://github.com/jazzband/django-silk/pull/667/files#diff-c6df0dfc9a321314dc99431481a130fe2136edb8d768cf86ad19ae4d9d8f3c15R92) as a fallback.

With that and the [custom filter manager](https://github.com/jazzband/django-silk/pull/667/files#diff-a273a37c67c5442c68d569b9adbc4b6d4297254c278addae900326f5683894cdR235) we can get filters from `session` or `silk_filters` if the first one is not defined.

What do you think?

Other question: would it be worth adding selenium tests?

**Comment by mgaligniana** [comment]
Hi @albertyw! Sorry to bother you, just wondering if there is something I can do to push forward the pull request. As I mentioned in my last comment, perhaps add selenium tests could give more confident to the change? Thank you!

**Review by albertyw — APPROVED** [review]

**Comment by MattyCZ** [comment]
Hi!

Sorry to bother, but any progress on the review? We would really appreciate this feature, since it would allow us to use silk without the middleware needed. Thank you very much

**Comment by mgaligniana** [comment]
Hi @MattyCZ!

This pull request was marked as part of the last release but never merged. I think we should wait.

I'm available in case of changes required!

**Comment by mgaligniana** [comment]
Hi @albertyw! I just want to let you know that this PR was marked as part of the release `5.1.0` but was never merged! https://github.com/jazzband/django-silk/releases/tag/5.1.0

**Pull request merged**